lib/manager/terraform/lockfile/index.spec.ts

```diff
@@ -0,0 +1,328 @@
+jest.mock('../../../datasource');
+
+jest.setTimeout(15000);
+
+const config = {
+  // `join` fixes Windows CI
+  localDir: join('/tmp/github/some/repo'),
+  cacheDir: join('/tmp/renovate/cache'),
+  constraints: {},
+};
+
+const validLockfile = loadFixture('validLockfile.hcl');
+
+const releaseBackendAWS = loadFixture('releaseBackendAWS_3_36_0.json');
+
+const releaseBackendRandom = loadFixture('releaseBackendRandom_3_1_0.json');
+
+const releaseBackendAzurerm = loadFixture('releaseBackendAzurerm_2_56_0.json');
+
+const serviceDiscoveryResult = loadFixture(
+  'service-discovery.json',
+  '../../../datasource/terraform-provider'
+);
+
+const fs: jest.Mocked<typeof _fs> = _fs as any;
+const mockHash = hash as jest.MockedFunction<typeof hash>;
+const mockGetPkgReleases = getPkgReleases as jest.MockedFunction<
+  typeof getPkgReleases
+>;
+
+const registryUrl = defaultRegistryUrls[0];
+const releaseBackendUrl = defaultRegistryUrls[1];
+
+describe(getName(), () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.resetModules();
+
+    httpMock.reset();
+    httpMock.setup();
+  });
+  afterEach(() => {
+    httpMock.reset();
+  });
+  it('returns null if no .terraform.lock.hcl found', async () => {
+    fs.readLocalFile.mockResolvedValueOnce(null);
+    expect(
+      await updateArtifacts({
+        packageFileName: 'main.tf',
+        updatedDeps: ['aws'],
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+  it('returns null if .terraform.lock.hcl is empty', async () => {
+    fs.readLocalFile.mockResolvedValueOnce('empty' as any);
+    expect(
+      await updateArtifacts({
+        packageFileName: 'main.tf',
+        updatedDeps: ['aws'],
+        newPackageFileContent: '',
+        config,
+      })
+    ).toBeNull();
+  });
+  it('update single dependency with exact constraint', async () => {
+    fs.readLocalFile.mockResolvedValueOnce(validLockfile as any);
+
+    mockHash.mockResolvedValueOnce([
+      'h1:lDsKRxDRXPEzA4AxkK4t+lJd3IQIP2UoaplJGjQSp2s=',
+      'h1:6zB2hX7YIOW26OrKsLJn0uLMnjqbPNxcz9RhlWEuuSY=',
+    ]);
+
+    httpMock
+      .scope(releaseBackendUrl)
+      .get('/terraform-provider-aws/3.36.0/index.json')
+      .reply(200, JSON.parse(releaseBackendAWS));
+
+    const localConfig: UpdateArtifactsConfig = {
+      updateType: 'minor',
+      newVersion: '3.36.0',
+      newValue: '3.36.0',
+      ...config,
+    };
+    const result = await updateArtifacts({
+      packageFileName: 'main.tf',
+      updatedDeps: ['hashicorp/aws'],
+      newPackageFileContent: '',
+      config: localConfig,
+    });
+    expect(result).not.toBeNull();
+    expect(result).toBeArrayOfSize(1);
+    expect(result[0].file).not.toBeNull();
+    expect(result[0].file).toMatchSnapshot();
+    expect(mockHash.mock.calls).toBeArrayOfSize(1);
+    expect(mockHash.mock.calls).toMatchSnapshot();
+  });
+  it('update single dependency with range constraint and minor update', async () => {
```

Comment:
```suggestion

  it('update single dependency with range constraint and minor update', async () => {
```

add newline for readabillity between tests.